pkg/util/compat/jsoncopy.go

```diff
@@ -0,0 +1,18 @@
+package compat
+
+import "encoding/json"
+
+// JSONCopy will copy src to dst via JSON serialization/deserialization.
+func JSONCopy(dst, src interface{}) error {
```

Comment:
Nice to have this in library! Can we add the unit test for this? In this case the more high level unit tests won't be necessary. Just ideas for checking - what happens if the dst has some fields not known to src and vice versa.

Reply:
This is actually well documented:
> By default, object keys which don't have a corresponding struct field are ignored (https://golang.org/pkg/encoding/json/#Unmarshal)

Thus, any mismatched fields/JSON keys will be silently omitted and left unchanged. (See https://goplay.space/#iPsozaeZY2W for an example)

That said, we're left with the following scenarios where this might fail:
1. Unconvertible types in dst/src with same name (i.e. `DiskSizeGB *int` <-> `DiskSizeGB *float64`)
2. Non-marshalable types in dst/src (i.e. `func()`)
3. Cyclic structs
4. Memory runs out

I find the 1st one to be at least somewhat relatable to our use cases, but we cannot test for it here.

Reply:
That said, we can add a sanity-check unit test for this function if the goal is to bump coverage to 100% :wink:

Reply:
thanks for the investigation above 👍 
I think it's good to have utility functions unit tested - this gives the confidence that no changes to them may break the main code. 
What do you think?